packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -1041,15 +1092,30 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
 ///
 ///  * [OverlayEntry], the imperative API for inserting a widget to an [Overlay].
 class OverlayPortal extends RenderObjectWidget {
-
-  /// Creates a widget that renders the given `overlayChild` on the [Overlay]
-  /// specified by `overlayChild`.
+  /// Creates a widget that renders the given [overlayChild] in the closest
+  /// ancestor [Overlay].
   const OverlayPortal({
     super.key,
-    required this.overlayInfo,
     required this.overlayChild,
     required this.child,
-  }) : assert(overlayChild == null || overlayInfo != null);
+  }) : _overlayInfo = null,
+       _overlayInfoGetter = OverlayInfo.maybeOf;
+
+  /// Creates a widget that renders the given [overlayChild] in the [Overlay]
+  /// specified by `overlayInfo`.
+  ///
+  /// The `overlayInfo` parameter must not be null when [overlayChild] is not
+  /// null.
+  const OverlayPortal.forOverlay({
+    super.key,
+    required OverlayInfo? overlayInfo,
+    required this.overlayChild,
+    required this.child,
+  }) : _overlayInfo = overlayInfo,
+       _overlayInfoGetter = _neverGetter,
+       assert(overlayChild == null || overlayInfo != null);
+
+  static Never _neverGetter(BuildContext context) => throw FlutterError('Unreachable!');
```

Comment:
Why can _overlayInfoGetter not just be nullable and null signals what _neverGetter does?